### ISSUE TITLE
Move multiple select from compat to core

### ISSUE THREAD

**Issue by marvinhagemeister** [developit/preact#1562]
In #1481 we proposed using the `value` attribute of a `select` element instead of setting the `selected` value on all options manually. But this means that setting multiple selected values is not possible. How should we solve this? In compat we added support for passing an array to `value` similar to how react handles it.

**Comment by marvinhagemeister** [comment]
Something the current solution in compat lacks is nested `optgroup` elements:

```jsx
<select>
    <optgroup label="Group 1">
        <option>A</option>
        <option>B</option>
        <option>C</option>
    </optgroup>
    <optgroup label="Group 2">
        <option>A2</option>
        <option>B2</option>
        <option>C2</option>
    </optgroup>
</select>
```

**Comment by natevw** [comment]
Looks like the proposed PR was closed since it wasn't close enough to "raw", which seems a reasonable decision if not also unfortunate. I happened to be revisiting my original code that led to my ticket at #1481 (± "closed as WONTFIX") which is still an open bug in the codebase on my end just low-priority to workaround. That is, I still have code that tries to set `selected=${state}` on each `<option>` and the recommendation for the **single** value case is to set the `value=${key}` on the `<select>`.

I'm bringing this up again here because I don't fully understand something which bears on the solution to  the **multiple** selection case:

> The reason is that keeping both the options selected-attributes and the select value in sync would require a bit of work and would cost us quite a bit in terms of size.

Can you unpack that? I.e. why is synchronization needed — won't the browser update the `selectEl.value` property itself when I would use preact to set the `optionEl.selected` property? Or maybe I should ask "where" is sync needed; is this more related to vdom diffing?

**Comment by marvinhagemeister** [comment]
Nah, it's more related edge cases in older browsers. Iirc there being many weird bugs in IE11 with `select.value` and only marking the options as `selected` seemed to work.
